### PR TITLE
klog-time-tracker: init at 6.2

### PR DESCRIPTION
**Pull request by blinry**
Hi! :) Excited to submit my first Nix package for a tool I love!

## Description of changes

klog is a command line tool for time tracking, which stores its data in a simple and human-readable plain-text format. klog is written in Go.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by blinry** [comment]
One thing I'm uncertain of whether `pname` should be "klog" (matches upstream) or "klog-time-tracker" (avoids naming conflicts with another package of the same name.

And the same question for the by-name path.

File: pkgs/by-name/kl/klog-time-tracker/package.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    sha256 = "1pj5hmqwba9h07npl4vpajns0768zg1dpxa7bhx1qwpc3av0ymiw";
+  };
+
+  vendorHash = "sha256-X5xL/4blWjddJsHwwfLpGjHrfia1sttmmqHjaAIVXVo=";
+
+  meta = with lib; {
+    description = "Command line tool for time tracking in a human-readable, plain-text file format";
+    homepage = "https://klog.jotaen.net";
+    license = licenses.mit;
+    maintainers = [maintainers.blinry];
```
Comment (AndersonTorres) [inline pkgs/by-name/kl/klog-time-tracker/package.nix:19]:
```suggestion
    maintainers = [ maintainers.blinry ];
```

File: pkgs/by-name/kl/klog-time-tracker/package.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{ buildGoModule, fetchFromGitHub, lib }:
+buildGoModule rec {
```
Comment (AndersonTorres) [inline pkgs/by-name/kl/klog-time-tracker/package.nix:2]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub }:

buildGoModule rec {
```

File: pkgs/by-name/kl/klog-time-tracker/package.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  pname = "klog-time-tracker";
+  version = "6.2";
+
+  src = fetchFromGitHub {
+    owner = "jotaen";
+    repo = "klog";
+    rev = "v${version}";
+    sha256 = "1pj5hmqwba9h07npl4vpajns0768zg1dpxa7bhx1qwpc3av0ymiw";
```
Comment (AndersonTorres) [inline pkgs/by-name/kl/klog-time-tracker/package.nix:10]:
```suggestion
    hash = "sha256-PFYPthrschw6XEf128L7yBygrVR3E3rtATCpxXGFRd4=";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
Result of `nixpkgs-review pr 270583` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>klog-time-tracker</li>
  </ul>
</details>

**Comment by AndersonTorres** [comment]
> One thing I'm uncertain of whether `pname` should be "klog" (matches upstream) or "klog-time-tracker" (avoids naming conflicts with another package of the same name.
> 
> And the same question for the by-name path.

Since there is another package with the same name, the new one should change somehow.

So you can name both `klog-time-tracker`

**Comment by blinry** [comment]
That's in fact how it's currently done! Ready to be merged, from my perspective!

**Review by AndersonTorres — APPROVED** [review]

**Comment by maralorn** [comment]
Cool, thank you!

**Comment by jotaen** [comment]
@blinry awesome, thanks for making this. Let me know if you’d like me to link this package from the [klog website](https://klog.jotaen.net/#get-klog).

**Comment by maralorn** [comment]
I am in favor.

**Comment by blinry** [comment]
@jotaen Of course, I'd love that! Thanks for building klog :)

**Comment by jotaen** [comment]
Done

**Pull request merged**